lio_sam:

  # Topics
  pointCloudTopic: "rslidar_points"
  imuTopic: "gnss_imu"
  odomTopic: "odometry/imu"                   # IMU pre-preintegration odometry, same frequency as IMU
  gpsTopic: "odometry/gps_manual"
  gpsRawTopic: "gps/fix"
  #FOR CHANGAN
  FOR_CHANGAN: true
  gps_CHANGAN_odom: "gnss_odom2"
  keyframenumber: 10000
  containersize: 1000

  # Frames
  lidarFrame: "base_link"
  baselinkFrame: "base_link"
  odometryFrame: "odom"
  mapFrame: "map"

  # GPS Settings
  useGps: false
  useImuHeadingInitialization: true           # if using GPS data, set to "true"
  useGpsElevation: true                     # if GPS elevation is bad, set to "false"
  gpsCovThreshold: 2.0                        # m^2, threshold for using GPS data
  poseCovThreshold: 0                      # m^2, threshold for using GPS data
  
  # Export settings
  savePCD: false                              # https://github.com/TixiaoShan/LIO-SAM/issues/3
  savePCDDirectory: "/Downloads/LOAM/"        # in your home folder, starts and ends with "/". Warning: the code deletes "LOAM" folder then recreates it. See "mapOptimization" for implementation

  # Save Path
  Odom_Path: "aaa/home/a/driver_ws/src/map_localization/results/"

  # Sensor Settings
  sensor: rslidar32                            # lidar sensor type, either 'velodyne' or 'ouster'
  N_SCAN: 32                                  # number of lidar channel (i.e., 16, 32, 64, 128)
  Horizon_SCAN: 1800                          # lidar horizontal resolution (Velodyne:1800, Ouster:512,1024,2048)
  downsampleRate: 1                           # default: 1. Downsample your data if too many points. i.e., 16 = 64 / 4, 16 = 16 / 1
  lidarMinRange: 2                          # default: 1.0, minimum lidar range to be used
  lidarMaxRange: 200.0                       # default: 1000.0, maximum lidar range to be used
  

  #特征点云地图（一张）+ SC重定位相关
  map_path: "/home/a/DATA/CS75_32new/CS75_MAPS/"  #if use sub map and big map with old maps
  # map_path: "/home/a/DATA/New_Map/"             #if use big map new version
  SubMapInfo: "/home/a/DATA/CS75_32new/CS75_MAPS/SubMapPose.txt"  #SC

  #GPS初始源(最新的地图)
  Renu: [0.994002, -0.022936, -0.106932,
         0.022567, 0.999734, -0.004658,
         0.107010, 0.002217, 0.994255]
  LLA_MAP: [39.9558622, 116.3105016, 45.2099098968]
  lever_arm: [0, 0, 0.905]
  
  #map_parameters
  UsagePointNum: 15000
  ndt_resolution: 0.5
  NumThreads: 4
  SearchMethod: "DIRECT7"
  transformepsilon: 1e-2
  StepSize: 0.01
  SubMapNum: 34
  KeySubMapNum: 1
  USE_SUBMAP: true

  # IMU Settings
  imuAccNoise: 3.9939570888238808e-03
  imuGyrNoise: 1.5636343949698187e-03
  imuAccBiasN: 6.4356659353532566e-05
  imuGyrBiasN: 3.5640318696367613e-05
  imuGravity: 9.805727
  imuRPYWeight: 0.01
  imuRate: 100.0
  imuQueSize: 1
  imuTimeOffset: -0.08

  # Extrinsics (IMU -> LiDAR)
  # extrinsicTrans: [0.76844, -0.0631438, 0.189212]

  # extrinsicTrans: [-0.82, 0.0, -1.4]

  extrinsicRot: [0.991601,-0.0450538, -0.12123,
                 0.0432943, 0.998916, -0.0171107,
                 0.121869, 0.0117184, 0.992477]
  extrinsicRPY: [0.991601,-0.0450538, -0.12123,
                 0.0432943, 0.998916, -0.0171107,
                 0.121869, 0.0117184, 0.992477]

  # extrinsicTrans: [0.0, 0.0, 0.0]

  # extrinsicRot: [1,0,0,0,1,0,0,0,1]
  # extrinsicRPY: [1,0,0,0,1,0,0,0,1]

  # LOAM feature threshold
  edgeThreshold: 1.0
  surfThreshold: 0.1
#   edgeThreshold: 0.5
#   surfThreshold: 0.001
  edgeFeatureMinValidNum: 10
  surfFeatureMinValidNum: 100
  USE_CORNER: true

  # extract feature threshold
  MaxCornerSize: 20 #20
  MaxSurfSize: 100 #60
  #Degenerate threshold
  Degenerate_Thr: 50.0
  MaxoptIteration: 15

  # voxel filter paprams
  odometrySurfLeafSize: 0.4                     # default: 0.4 - outdoor, 0.2 - indoor
  mappingCornerLeafSize: 0.2                    # default: 0.2 - outdoor, 0.1 - indoor
  mappingSurfLeafSize: 0.4                      # default: 0.4 - outdoor, 0.2 - indoor

  # ieskf on manifold
  LIDAR_STD: 0.001
  USE_S2: false

  # robot motion constraint (in case you are using a 2D robot)
  z_tollerance: 1000                          # meters
  rotation_tollerance: 1000                     # radians

  # CPU Params
  numberOfCores: 4                              # number of cores for mapping optimization
  mappingProcessInterval: 0.15                  # seconds, regulate mapping frequency

  # Surrounding map
  surroundingkeyframeAddingDistThreshold: 1.0   # meters, regulate keyframe adding threshold
  surroundingkeyframeAddingAngleThreshold: 0.1  # radians, regulate keyframe adding threshold
  surroundingKeyframeDensity: 2.0               # meters, downsample surrounding keyframe poses   
  surroundingKeyframeSearchRadius: 50.0         # meters, within n meters scan-to-map optimization (when loop closure disabled)

  # Loop closure
  loopClosureEnableFlag: true
  loopClosureFrequency: 1.0                     # Hz, regulate loop closure constraint add frequency
  surroundingKeyframeSize: 50                   # submap size (when loop closure enabled)
  historyKeyframeSearchRadius: 15.0             # meters, key frame that is within n meters from current pose will be considerd for loop closure
  historyKeyframeSearchTimeDiff: 30.0           # seconds, key frame that is n seconds older will be considered for loop closure
  historyKeyframeSearchNum: 25                  # number of hostory key frames will be fused into a submap for loop closure
  historyKeyframeFitnessScore: 0.3              # icp threshold, the smaller the better alignment

  # Visualization
  globalMapVisualizationSearchRadius: 1000.0    # meters, global map visualization radius
  globalMapVisualizationPoseDensity: 10.0       # meters, global map visualization keyframe density
  globalMapVisualizationLeafSize: 1.0           # meters, global map visualization cloud density